integration_tests/src/main/python/cast_test.py

```diff
@@ -167,3 +166,16 @@ def test_cast_long_to_decimal_overflow():
             f.col('a').cast(DecimalType(18, -1))),
         conf={'spark.sql.legacy.allowNegativeScaleOfDecimal': True})
 
+
+@pytest.mark.xfail(not is_before_spark_311(), reason="Only in Spark 3.1.1+ do we have incompatibility with negative decimal scale", raises=IllegalArgumentException)
```

Comment:
If an xfail passes it does not fail it is an XPass that everyone ignores. I would prefer for us to write tests that are much more specific instead.  Is it returning the wrong value? is not on the GPU. What is supposed to actually be happening in these cases?

Reply:
Basically what I am trying to achieve is that if the Spark version is >= 311 then this test should throw an IllegalArgument exception for this to xfail.

Reply:
>  if the Spark version is >= 311 then this test should throw an IllegalArgument exception for this to xfail.

Do we expect this to ever change (i.e.: is there a bug in the plugin or in Spark that we expect to get fixed in the future such that this test would pass)?  xfail should be reserved for cases where we eventually expect them to pass.  We should not be xfailing tests that we never expect to pass, as that just creates noise in the test summary output and wastes test bandwidth.  Instead we should be skipping such configurations for the test since we know the test premise is invalid for that setup.